### PR TITLE
wandio: add dependencies and update license

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/wandio.rb [hunk 1]
```diff
@@ -18,6 +19,15 @@ class Wandio < Formula
     sha256 cellar: :any, high_sierra:   "29602aec2851811108e97397e6310f091f5e5fe0844f9cfcd6657193d9c53ff4"
   end
 
+  depends_on "lz4"
+  depends_on "lzo"
+  depends_on "xz" # For LZMA
+  depends_on "zstd"
```
Comment (cho-m) [inline Formula/wandio.rb:25]: At least description claims support for lzma and zstd, but these are not available unless dependencies are available at compile-time (`xz` and `zstd`). Not sure if worth enabling other formats (`lz4` and `lzo`), but added for now.

Existing bottles are probably built with:
```
configure: Compiled with compressed file (zlib) support: Yes
configure: Compiled with compressed file (bz2) support: Yes
configure: Compiled with compressed file (lzo write only) support: No
configure: Compiled with compressed file (lzma) support: No
configure: Compiled with compressed file (zstd) support: No
configure: Compiled with compressed file (lz4) support: No
configure: Compiled with Intel QuickAssist Technology support: No
configure: Compiled with http read (libcurl) support: Yes
```

With dependencies:
```
configure: Compiled with compressed file (zlib) support: Yes
configure: Compiled with compressed file (bz2) support: Yes
configure: Compiled with compressed file (lzo write only) support: Yes
configure: Compiled with compressed file (lzma) support: Yes
configure: Compiled with compressed file (zstd) support: Yes
configure: Compiled with compressed file (lz4) support: Yes
configure: Compiled with Intel QuickAssist Technology support: No
configure: Compiled with http read (libcurl) support: Yes
```

File: Formula/wandio.rb [hunk 2]
```diff
@@ -3,7 +3,8 @@ class Wandio < Formula
   homepage "https://research.wand.net.nz/software/libwandio.php"
   url "https://research.wand.net.nz/software/wandio/wandio-4.2.3.tar.gz"
   sha256 "78c781ce2c3783b85d894e29005b7e98fc246b33f94616047de3bb4d11d4d823"
-  license "GPL-3.0"
+  license "LGPL-3.0-or-later"
```
Comment (cho-m) [inline Formula/wandio.rb:6]: License according to homepage:

> **NEW: From libwandio 1.0.4 onwards, the libwandio source code is licensed under the LGPL v3.**
> 
> All releases of libwandio prior to 1.0.4 are licensed under the GPL v2.

Though some of files still show GPL-2.0 or later

File: Formula/wandio.rb (hunk 1 above)
Comment (carlocab) [reply Formula/wandio.rb:25]: Do we want all the extra dependencies?

Comment (cho-m) [reply Formula/wandio.rb:25]: Not sure. Wandio's main feature is dealing with multiple compression formats and their installation recommendation is to add all libraries https://github.com/wanduow/wandio/blob/master/INSTALL
```
Strongly recommended:
 * libpthread
 * zlib-dev
 * libbz2-dev
 * liblzma-dev
 * liblzo2-dev
 * liblz4-dev
 * libzstd-dev
```

I assume they do so for their maintained packages:
- https://github.com/wanduow/wandio/wiki/Installing-via-RPM
- https://github.com/wanduow/wandio/wiki/Installing-via-Debian-Package

File: Formula/wandio.rb (hunk 2 above)
Comment (carlocab) [reply Formula/wandio.rb:6]: Do they indicate whether it's `-or-later` or `-only`?

Comment (cho-m) [reply Formula/wandio.rb:6]: Only saw it in parts of available code, e.g.
https://github.com/wanduow/wandio/blob/4.2.3-1/lib/curl-helper.c#L11-L14
```
 * libwandio is free software; you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
```

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**